AuctionSite.Bl/User/RegisterUserLogic.cs

```diff
@@ -0,0 +1,77 @@
+﻿using AuctionSite.BL.Common;
+using AuctionSite.BL.User.Interfaces;
+using AuctionSite.BL.User.Internal;
+using AuctionSite.BL.User.Validators;
+using AuctionSite.DataAccess;
+using AuctionSite.Shared;
+using AuctionSite.Shared.BindingModel;
+using AuctionSite.Shared.Dto;
+using AuctionSite.Shared.Enums;
+using AuctionSite.Shared.Exceptions;
+using AuctionSite.Shared.ViewModel;
+using AutoMapper;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AuctionSite.BL.User
+{
+    internal class RegisterUserLogic : BaseBusinessLogic<RegisterUserBindingModel, UserDto, RegisterUserViewModel>, IRegisterUserLogic
+    {
+        public RegisterUserLogic(
+            IUnitOfWork unitOfWork,
+            IMapper mapper, 
+            ILogger<RegisterUserLogic> logger) 
+            : base(unitOfWork, mapper, logger)
+        {
+        }
+
+        protected override IEnumerable<IValidator<RegisterUserBindingModel>> Validators =>
+
+            new IValidator<RegisterUserBindingModel>[]
+            {
+                new EmailFormatValidator(),
+                new UserExistenceValidator(UnitOfWork),
+                new PasswordValidator()
+            };
+
+        protected override async Task<UserDto> ExecutionAsync(RegisterUserBindingModel registerUserBindingModel)
+        {
+            var userDto = MapUser(registerUserBindingModel);
+
+            await UnitOfWork.User.AddUser(userDto);
+            return userDto;
+        }
+
+        private UserDto MapUser(RegisterUserBindingModel registerUserBindingModel)
+        {
+            var user = Mapper.Map<UserDto>(registerUserBindingModel);
+            user.PasswordHash = GetHash(registerUserBindingModel.Password);
+            user.Role = Role.User;
+            user.RecoveryGuid = Guid.NewGuid();
+            user.CreationDate = DateTime.UtcNow;
+
+            return user;
+        }
+
+        private string GetHash(string password)
```

Comment:
Myślę, że możemy to śmiało umieść gdzieś w shared jako extension, ponieważ będziemy korzystać z hashowania jeszcze

Reply:
Możemy